packages/matter.js/src/protocol/interaction/InteractionServer.ts

```diff
@@ -371,17 +388,248 @@ export function attributePathToId({ endpointId, clusterId, attributeId }: TypeFr
     return `${endpointId}/${clusterId}/${attributeId}`;
 }
 
-function toHex(value: number | undefined) {
-    return value === undefined ? "*" : `0x${value.toString(16)}`;
+export function eventPathToId({ endpointId, clusterId, eventId }: TypeFromSchema<typeof TlvEventPath>) {
+    return `${endpointId}/${clusterId}/${eventId}`;
 }
 
-export class InteractionServer implements ProtocolHandler<MatterDevice> {
+export class EndpointStructure {
+    endpoints = new Map<number, Endpoint>();
+    attributes = new Map<string, (AttributeServer<any> | FabricScopedAttributeServer<any> | FixedAttributeServer<any>)>();
+    attributePaths = new Array<AttributePath>();
```

Comment:
I'm wondering if attributes could be saved as AttributeWithPath?  Would make state management simpler and the search code wouldn't have to do the secondary lookup from attributePaths -> attributes[attributePathToId(path)]

Reply:
AttributeWIthPath would bring no benefit becauseit would be in the data. with that "string key that contains "endpointId/clusterId/attributeId" is a direct match ... else you would need to search any time

Reply:
Right, I didn't mean to imply get rid of the string key, just have it point to AttributeWithPath instead of Attribute

Reply:
But Attributeath is something different then AttributeServerWithPath :-) so another set of types ... would prefer to leave it for now